src/unix/syscall.c

```diff
@@ -1076,11 +1006,16 @@ sysreturn getdents64(int fd, struct linux_dirent64 *dirp, unsigned int count)
 
 sysreturn chdir(const char *path)
 {
+    int ret;
     tuple n;
     if (path == 0)
         return set_syscall_error(current, EINVAL);
```

Comment:
Should be EFAULT, I think...can actually remove this clause as resolve_cstring_follow should check anyway, right?

Reply:
Right, will remove this clause since resolve_cstring_follow() does the check for us.